packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/common/useMemberData.ts

```diff
@@ -0,0 +1,68 @@
+import app from 'state';
+import { useFetchGroupsQuery } from 'state/api/groups';
+import { Memberships } from 'state/api/groups/refreshMembership';
+import { trpc } from 'utils/trpcClient';
+import { CWTableState } from 'views/components/component_kit/new_designs/CWTable/useCWTableState';
+
+interface UseMemberDataProps {
+  memberships?: Memberships[];
+  tableState?: CWTableState;
+  groupFilter?: string;
+  debouncedSearchTerm?: string;
+  membersPerPage: number;
+  page: number;
+  allowedAddresses: string[];
+}
+
+export const useMemberData = ({
+  memberships,
+  tableState,
+  groupFilter,
+  debouncedSearchTerm,
+  membersPerPage,
+  page,
+  allowedAddresses,
+}: UseMemberDataProps) => {
+  const parseMembership = !isNaN(groupFilter as unknown as number)
+    ? `in-group:${groupFilter}`
+    : groupFilter;
+
+  const membershipsFilter = ['all-community'].includes(groupFilter.toString())
+    ? undefined
+    : parseMembership;
+
+  const { data: members, isLoading: isLoadingMembers } =
+    trpc.community.getMembers.useQuery(
```

Comment:
despite trpc, we would like to keep all FE <=> BE interface in one place which is `state/api`. The reason for this is to have everything in one place, no matter if we use trpc, react-query or axios. To adjust this, please:
- create `getMembers.ts` file in `state/api/members`
- create `useGetMembers` hook inside and export it 
- use `useGetMembers` hook here in this file

plase use `useGetContestsQuery` as a reference